cmd/lotus-shed/miner.go

```diff
@@ -354,3 +361,220 @@ var minerUnpackInfoCmd = &cli.Command{
+	Flags: []cli.Flag{
+		&cli.BoolFlag{
+			Name:  "really-do-it",
+			Usage: "Actually send transaction performing the action",
+			Value: false,
+		},
+		&cli.Int64SliceFlag{
+			Name:     "partitions",
+			Usage:    "list of partitions to compact sectors in",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:  "actor",
+			Usage: "TODO",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		if !cctx.Bool("really-do-it") {
+			return xerrors.Errorf("Pass --really-do-it to actually execute this action")
+		}
+
+		api, acloser, err := lcli.GetFullNodeAPI(cctx)
+		if err != nil {
+			return xerrors.Errorf("getting api: %w", err)
+		}
+		defer acloser()
+
+		ctx := lcli.ReqContext(cctx)
+
+		maddr, err := address.NewFromString(cctx.String("actor"))
+		if err != nil {
+			return xerrors.Errorf("getting actor address: %w", err)
+		}
+
+		minfo, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+		if err != nil {
+			return xerrors.Errorf("getting mienr info: %w", err)
+		}
+
+		deadline, err := api.StateMinerProvingDeadline(ctx, maddr, types.EmptyTSK)
+		if err != nil {
+			return xerrors.Errorf("getting deadline: %w", err)
+		}
+
+		partitionIndices := cctx.Int64Slice("partitions")
+		if len(partitionIndices) <= 0 {
+			return fmt.Errorf("must include at least one partition to compact")
+		}
+
+		chainHead, err := api.ChainHead(ctx)
+		if err != nil {
+			return xerrors.Errorf("getting chain head: %w", err)
+		}
+
+		checkRand, err := api.StateGetRandomnessFromTickets(ctx, crypto.DomainSeparationTag_PoStChainCommit, deadline.Challenge, nil, chainHead.Key())
+		if err != nil {
+			return xerrors.Errorf("getting randomness: %w", err)
+		}
+
+		proofSize, err := minfo.WindowPoStProofType.ProofSize()
+		if err != nil {
+			return xerrors.Errorf("getting proof size: %w", err)
+		}
+
+		var partitions []miner8.PoStPartition
+
+		emptyProof := []proof.PoStProof{{
+			PoStProof:  minfo.WindowPoStProofType,
+			ProofBytes: make([]byte, proofSize)}}
+
+		for _, partition := range partitionIndices {
+			newPartition := miner8.PoStPartition{
+				Index:   uint64(partition),
+				Skipped: bitfield.New(),
+			}
+			partitions = append(partitions, newPartition)
+		}
+
+		params := miner8.SubmitWindowedPoStParams{
+			Deadline:         deadline.Index,
+			Partitions:       partitions,
+			Proofs:           emptyProof,
+			ChainCommitEpoch: deadline.Challenge,
+			ChainCommitRand:  checkRand,
+		}
+
+		sp, err := actors.SerializeParams(&params)
+		if err != nil {
+			return xerrors.Errorf("serializing params: %w", err)
+		}
+
+		fmt.Printf("submitting bad PoST for %d paritions\n", len(partitionIndices))
+		smsg, err := api.MpoolPushMessage(ctx, &types.Message{
+			From:   minfo.Worker,
+			To:     maddr,
+			Method: builtin.MethodsMiner.SubmitWindowedPoSt,
+			Value:  big.Zero(),
+			Params: sp,
+		}, nil)
+		if err != nil {
+			return xerrors.Errorf("mpool push: %w", err)
+		}
+
+		fmt.Println("Message CID:", smsg.Cid())
+
+		return nil
+	},
```

Comment:
Would also wait for inclusion and success here -- if you're running this command, you're definitely gonna wanna make sure it succeeds, in which case you might as well do it as part of the command.